main.tf

```diff
@@ -13,8 +13,13 @@ locals {
   private_key_filename = "${var.ssh_public_key_path}/${module.label.id}${var.private_key_extension}"
 }
 
+resource "null_resource" "ASSERTION_FAILED" {
+  count                                                                     = "${var.generate_ssh_key == "true" && var.import_ssh_key == "true" ? 1 : 0}"
+  "You cannot generate AND import a key, you can only do one or the other." = true
```

Comment:
@Jamie-BitFlight 
@osterman 
let's discuss and finish this PR.

Two things that I don't really like:
1. That kind of `assertion` above, which looks like a hack and will confuse many people
2. What the `assertion` says: "You cannot generate AND import a key, you can only do one or the other"

But this is actually what we always do in our projects - we generate an SSH key and then import it into AWS.
There is an option to not generate a key, but just import an existing one.
This module was not supposed to be used just to generate a key (we created a diff module for that https://github.com/cloudposse/terraform-tls-ssh-key-pair).

@Jamie-BitFlight 
I think you are referring to a different issue here.
If you use `file`, Terraform will always try to read the file even if it does not exist or is not in use.
(we were placing a `dummy` file before to get around this issue).